### ISSUE TITLE
Documentation for derive macros is in the wrong place

### ISSUE THREAD

**Issue by JoJoJet** [bevyengine/bevy#6195]
## How can Bevy's documentation be improved?

For derivable traits such as `WorldQuery`, `SystemParam`, `Bundle`, etc., documentation for the derive macro is placed on the [trait definition](https://github.com/bevyengine/bevy/blob/6b75589e2c91429eb6c35574840d47d39b2379b9/crates/bevy_ecs/src/query/fetch.rs#L46) itself, rather than the derive macro. This means that when hovering over a `#[derive(WorldQuery)]` attribute, it appears to be [mostly undocumented](https://github.com/bevyengine/bevy/blob/6b75589e2c91429eb6c35574840d47d39b2379b9/crates/bevy_ecs/macros/src/lib.rs#L403).

Any documentation regarding the derive macro should be moved *to* the derive macro -- this information is not relevant to consumers of the trait, and is not discoverable for those trying to derive it.

**Comment by JayPavlina** [comment]
I am working on this, but it will require circular dev dependencies for the doc tests to pass. Is that ok?

**Comment by JoJoJet** [comment]
Ah, that's unfortunate. I don't imagine there's any good workarounds for that issue -- this idea might not be worth pursuing :(

**Comment by JayPavlina** [comment]
The only workaround I've seen mentioned is to manually link to docs.rs, but that doesn't seem like a good solution to me.

**Comment by JoJoJet** [comment]
I agree. I'd love to move the docs to a better place, but IMO it's not worth using ugly workarounds.

**Comment by ItsDoot** [comment]
This issue goes beyond just the ECS now. `Asset`, `States`, and `SubStates` are also missing documentation.
